src/sliding_sync.rs

```diff
@@ -887,6 +941,37 @@ pub fn take_timeline_endpoints(
 }
 
 
+/// Return an option of user's fully read event id and timestamp for a given room using Mutex Lock
+/// 
+/// Gets the fully read event for the given room
+/// Returns `None` if there is no fully read event
+pub fn take_fully_read_event(room_id: &OwnedRoomId) -> Option<(OwnedEventId, MilliSecondsSinceUnixEpoch)> {
```

Comment:
Please rename this function, since this isn't _taking_ the event, it's just getting it and then returning a copy of it.

In Rust terminology, to `take` something means to remove the existing value and return it, replacing the existing value with the default. So when you "take" an option, it will always be `None` afterwards.


```suggestion
pub fn get_fully_read_event(room_id: &OwnedRoomId) -> Option<(OwnedEventId, MilliSecondsSinceUnixEpoch)> {
```